### PR TITLE
Disable KD mode from saving problematic state

### PR DESCRIPTION
**Pull request by AAnoosheh**
## What does this PR do?

Feature: Change the way "kd_loss" mode saves state

**Overview:** ?

## Usage

```python
# Add a code snippet demonstrating how to use this
```

## Testing
Unit

## Before your PR is "*Ready for review*"
- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Y (technically No in theory, but in practice yes)
- **Did you write any new necessary tests?**: Y
- **Did you add or update any necessary documentation?**: N
- **Did you update [Changelog](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CHANGELOG.rst)?**: N

## Summary by CodeRabbit

- New Features
  - Distillation now accepts a pre-instantiated teacher model (nn.Module) in addition to class, callable, or tuple forms.

- Refactor
  - Checkpoint restore no longer reinstates the distillation wrapper; exported students restore with KD config reset and a new pre-save reset hook.
  - Saved state now preserves all KD-related entries (no pruning).

- Documentation
  - Guides and examples updated for teacher_model forms, export/restore semantics, loss kwargs, CLI formatting, arg rename (max_seq_length → max_length), and example requirements.

- Tests
  - Tests updated to assert unrestored distillation state and adjust examples/tests to direct teacher instantiation.

### CODE REVIEW

File: examples/llm_distill/requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 pyarrow
-trl==0.13.0
+trl==0.23.0
```
Comment (kevalmorabia97) [inline examples/llm_distill/requirements.txt]: Any reason to pin a specific version instead of `>=`?
```suggestion
trl>=0.23.0
```

File: examples/llm_distill/README.md [hunk 2]
```diff
@@ -158,35 +154,47 @@ Keep in mind the training loss of the distillation run is not directly comparabl
 ### Train teacher
 
 ```bash
-accelerate launch --multi_gpu --mixed_precision bf16  main.py \
+accelerate launch \
```
Comment (kevalmorabia97) [inline examples/llm_distill/README.md]: Can you trigger llm_distill example test in internal CI to make sure it passes with FSDP 2?

File: examples/llm_distill/main.py [hunk 3]
```diff
@@ -128,14 +120,13 @@ def train():
...
+        )
         kd_config = {
-            "teacher_model": (
-                _teacher_factory,
-                (model_args.teacher_name_or_path,),
-                {},
-            ),
+            "teacher_model": teacher_model,
             "criterion": LMLogitsLoss(),
-            "expose_minimal_state_dict": False,  # FSDP forces us to disable this
```
Comment (AAnoosheh) [inline examples/llm_distill/main.py:138]: @kinjalpatel27 We can also eventually move all the other examples from FSDP1 to FSDP2 (which does not have this state dict requirement)

File: examples/llm_distill/requirements.txt (hunk 1 above)
Comment (realAsma) [reply examples/llm_distill/requirements.txt]: +1

File: examples/llm_distill/main.py [hunk 4]
```diff
@@ -117,7 +107,9 @@ def train():
 
     if model_args.single_model:
         logger.info("Loading single model only...")
-        model = _teacher_factory(model_path)
+        model = transformers.AutoModelForCausalLM.from_pretrained(
+            model_path, device_map=PartialState().process_index
```
Comment (realAsma) [inline examples/llm_distill/main.py]: why is `device_map` needed here? Without the device map, the model will be loaded in CPU (indistributed mode, all ranks except rank 0 will be loaded in meta device).
 ```suggestion
            model_path, torch_dtype=torch.bfloat16
```

File: examples/llm_distill/README.md [hunk 5]
```diff
@@ -158,35 +154,47 @@ Keep in mind the training loss of the distillation run is not directly comparabl
...
-    --fsdp 'full_shard auto_wrap' \
-    --fsdp_transformer_layer_cls_to_wrap LlamaDecoderLayer
+    --gradient_checkpointing True
 ```
 
 ### Distill teacher into student
 
 ```bash
-accelerate launch --multi_gpu --mixed_precision bf16  main.py \
+accelerate launch \
```
Comment (kinjalpatel27) [inline examples/llm_distill/README.md]: nitpick: it might be easier to use config file for accelerate launch instead of passing individual parameters, similar to llm_qat example

File: examples/llm_distill/README.md (hunk 2 above)
Comment (kevalmorabia97) [reply examples/llm_distill/README.md]: ‼️ Failing https://gitlab-master.nvidia.com/omniml/modelopt/-/jobs/210706525

File: examples/llm_distill/requirements.txt (hunk 1 above)
Comment (AAnoosheh) [reply examples/llm_distill/requirements.txt]: Every version they might randomly change argument names. Like since the last one they changed `max_seq_length` to `max_length` and before that they changed several others.

File: examples/llm_distill/README.md (hunk 2 above)
Comment (AAnoosheh) [reply examples/llm_distill/README.md]: Oh right forgot to actually update the test folder script itself

File: examples/llm_distill/main.py (hunk 4 above)
Comment (AAnoosheh) [reply examples/llm_distill/main.py]: It was needed back when I first wrote this example else it didn't load properly. Nowadays maybe it works without it.

File: examples/llm_distill/requirements.txt (hunk 1 above)
Comment (AAnoosheh) [reply examples/llm_distill/requirements.txt]: Changed it but let's see what happens in future

File: examples/llm_distill/README.md (hunk 2 above)
Comment (AAnoosheh) [reply examples/llm_distill/README.md]: Fixed: https://gitlab-master.nvidia.com/omniml/modelopt/-/jobs/210853166

**Review by kevalmorabia97 — APPROVED** [review]
